### PR TITLE
[ci]: Set up CI to automate WinGet submissions

### LINKED ISSUE #1217: Automate submission of new releases to WinGet
Since the release process of DevHome has been simplified into releasing an msixbundle and we have a [base manifest](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/DevHome/) over at winget-pkgs, a CI can be setup that can automate new submissions to the WinGet repository using [winget-create](https://github.com/microsoft/winget-create). This will ensure that DevHome versions are up-to-date in WinGet repository.

For implementation, we can take a look at how PowerToys has set-up the CI https://github.com/microsoft/PowerToys/blob/main/.github/workflows/package-submissions.yml

### PR DESCRIPTION
**Pull request by mdanish-kh**
## Summary of the pull request
Add winget-submission.yml to automatically submit a newer version to winget-pkgs repository for a new release. Inspiration taken from https://github.com/microsoft/PowerToys/blob/main/.github/workflows/package-submissions.yml

## References and relevant issues
- #1217

## Detailed description of the pull request / Additional comments

For the CI to work, a maintainer needs to add a GitHub Personal-Access-Token (PAT) as a "Repository Secret" named `WINGET_PAT`. Instructions on creating a PAT can be found over at [winget-create's repo](https://github.com/microsoft/winget-create#github-personal-access-token-classic-permissions)

## Validation steps performed

Tested all the commands locally; The following command be used as a test on how a hypothetical update would work using winget-create (Don't hesitate on running the command as it won't generate and auto-submit a PR, I've omitted the `--submit` flag from it)

```powershell
wingetcreate update Microsoft.DevHome --version 2.0 --urls "https://github.com/microsoft/devhome/releases/download/v0.200.170.0/Windows.DevHome_0.200.170.0.msixbundle"
```

## PR checklist
- [x] Closes #1217
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

**Comment by AmelBawa-msft** [comment]
Looks good to me. Adding @EricJohnson327 to help with the repo secret bit.

**Comment by denelon** [comment]
@ryfu-msft as another set of eyes here as well.

File: .github/workflows/winget-submission.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+          # Fetching latest release from GitHub
+          $github = Invoke-RestMethod -uri "https://api.github.com/repos/microsoft/devhome/releases"
+          $targetRelease = $github | Where-Object -Property name -match 'Dev Home'| Select-Object -First 1
+          $installerUrl = $targetRelease | Select-Object -ExpandProperty assets -First 1 | Where-Object -Property name -match 'Windows.DevHome.*?msixbundle' | Select-Object -ExpandProperty browser_download_url
+          $packageVersion = $targetRelease.tag_name.Trim("v")
+
+          # Update package using wingetcreate
+          Invoke-WebRequest https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
+          .\wingetcreate.exe update $packageId --version $packageVersion --urls "$installerUrl" --submit --token $gitToken
```
Comment (ryfu-msft) [inline .github/workflows/winget-submission.yml:27]: Nit: Variables should enclosed in parentheses like `$(packageId)`. That's what we do for wingetcreate and the explanation is here: https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#macro-syntax-variables

**Comment by EricJohnson327** [comment]
Thanks @mdanish-kh, this looks great!  I'm new to GitHub Actions and WinGet so I want to make sure this works as expected.  Today, we build Dev Home in ADO and publish to the Store from there.  Is this the expected flow now when we do a release?  I just want to make I'm not missing anything!

1) Run ADO pipeline and Store publish
2) Add release packages to GitHub releases page
3) Run winget-submission.yml to update WinGet with the package from step 2

**Review by ryfu-msft — APPROVED** [review]

File: .github/workflows/winget-submission.yml (hunk 1 above)
Comment (mdanish-kh) [reply .github/workflows/winget-submission.yml:27]: Prior to creating a PR here, I was testing this workflow in a private fork; It seems when I add $(packageId), it doesn't seem to work
image

I'll make the fork public if you want to see the details

**Comment by mdanish-kh** [comment]
> Thanks @mdanish-kh, this looks great! I'm new to GitHub Actions and WinGet so I want to make sure this works as expected. Today, we build Dev Home in ADO and publish to the Store from there. Is this the expected flow now when we do a release? I just want to make I'm not missing anything!
> 
>     1. Run ADO pipeline and Store publish
> 
>     2. Add release packages to GitHub releases page
> 
>     3. Run winget-submission.yml to update WinGet with the package from step 2

@EricJohnson327 The CI will automatically kick off when you do step 2 i.e., publish a new release on GitHub. There's no need to do step 3

**Comment by EricJohnson327** [comment]
> > Thanks @mdanish-kh, this looks great! I'm new to GitHub Actions and WinGet so I want to make sure this works as expected. Today, we build Dev Home in ADO and publish to the Store from there. Is this the expected flow now when we do a release? I just want to make I'm not missing anything!
> > ```
> > 1. Run ADO pipeline and Store publish
> > 
> > 2. Add release packages to GitHub releases page
> > 
> > 3. Run winget-submission.yml to update WinGet with the package from step 2
> > ```
> 
> @EricJohnson327 The CI will automatically kick off when you do step 2 i.e., publish a new release on GitHub. There's no need to do step 3

I like it!  Thank you

**Review by EricJohnson327 — APPROVED** [review]

File: .github/workflows/winget-submission.yml (hunk 1 above)
Comment (ryfu-msft) [reply .github/workflows/winget-submission.yml:27]: Ah that's interesting, I would imagine they would be similar but I guess not!

**Comment by mdanish-kh** [comment]
@EricJohnson327 I would suggest running it manually once as a test to see if the github secret is being referenced properly or not. Ofcourse this will generate a PR over at winget-pkgs, but you / or a mod can shortly close it after its creation

**Pull request merged**